olive/passes/onnx/conversion.py

```diff
@@ -75,8 +75,10 @@ def _run_for_config(
 
         # TODO: add e2e test for model on cpu but data on gpu; model on gpu but data on cpu
         # put pytorch_model and dummy_inputs at the same device
-        pytorch_model.to("cpu")
-        dummy_inputs = tensor_data_to_device(dummy_inputs, "cpu")
+        # device = 'cpu' if self.accelerator_spec.accelerator_type == Device.CPU else 'cuda'
+        device = device_string_to_torch_device(self.accelerator_spec.accelerator_type)
+        pytorch_model.to(device)
```

Comment:
Does the final model differ whether we use the `cpu` or `gpu`? Or is this just to speed up the model forward pass during tracing?

Reply:
Support for GPU based export is required because certain operators are implemented only for GPU and the export fails if attempted on CPU.

I can't compare the model output because the export fails on CPU because of lack of supported operators. Export performance was not a criteria for this change.